### ISSUE TITLE
Let graphql query with preview token accept id/uid to preview draft entry

### ISSUE THREAD

**Issue by membla** [craftcms/cms#5317]
### Description

In a headless preview scenario, when using a graphql entry query that selects an entry based on id/uid, passing the preview token will not return the draft version since the draft version will have a different id/uid. Would it be possible for id/uid+token to return the draft entry? If not, could the graphql entry query accept a sourceId param? Otherwise, as far as I can see we are limited to using slugs in the query to identify the entry during preview, which at best seems non-obvious.

### Additional info

- Craft version: 3.4.
- PHP version: 7.3.11
- Database driver & version: MySQL 5.7.28
- Plugins & versions: -

**Comment by brandonkelly** [comment]
This already does work, and I just verified it. I’m able to query for an entry by either its ID or UID with GraphQL, and if the correct `token` param is in the URL, then the entry’s draft content will be returned instead.

**Comment by denisyilmaz** [comment]
@brandonkelly we use angular for our headless frontend framework. We have a routing call on startup which gets all routes from CraftCMS and creates a route-mapping with ID and siteID, which works great except for the preview.

For this we have an interceptor which checks if the http request is going to the GraphQL endpoint of CraftCMS. If so, it checks the called url for a `x-craft-token` param and then adds the `token` param to the api request.
As we will append the token to all `/api` requests, our routing call will have the `token`param as well which results in all routes having the id set to the draftID, not the original ID of the entry.

```gql
# Default Page
query GetPage($id: QueryArgument!, $siteId: QueryArgument!) {
  entry(id: [$id], siteId: [$siteId]) {
    id
    siteId
    title
    slug
    uri
    url
    typeHandle
    sectionHandle
    language
  }
}

```
will fail when the `$id` is set to the draft ID + token. resulting in an empty entry response:
```json
{
  "data": {
    "entry": null
  }
}
```

so basically the `token` preview fails if ID is correclty set to the Draft-ID.
Is there a way to get this to work as expected?
